sdk/java-sdk-spring/src/main/scala/kalix/javasdk/client/ComponentCall.scala

```diff
@@ -107,6 +111,24 @@ object ComponentCall {
     }
   }
 
+  private def getReturnType[R](declaringClass: Class[_], method: Method): Class[R] = {
+    if (classOf[Action].isAssignableFrom(declaringClass)
+      || classOf[ValueEntity[_]].isAssignableFrom(declaringClass)
+      || classOf[EventSourcedEntity[_, _]].isAssignableFrom(declaringClass)
+      || classOf[Workflow[_]].isAssignableFrom(declaringClass)) {
+      //here we are expecting a wrapper in the form of an Effect
+      method.getGenericReturnType.asInstanceOf[ParameterizedType].getActualTypeArguments.head.asInstanceOf[Class[R]]
+    } else {
+      //in other cases we expect a View query method, but declaring class could not extend View[_] class for join views
```

Comment:
```suggestion
      // in other cases we expect a View query method, but declaring class may not extend View[_] class for join views
```

otherwise is like it was not able to extend